extensions/kogito/deployment/src/main/java/io/quarkus/kogito/deployment/KogitoAssetsProcessor.java

```diff
@@ -69,16 +131,50 @@ public void generateModel(ArchiveRootBuildItem root,
         Collection<GeneratedFile> generatedFiles = appGen.generate();
 
         if (!generatedFiles.isEmpty()) {
+
+            Indexer kogitoIndexer = new Indexer();
+            Set<DotName> kogitoIndex = new HashSet<>();
+
             MemoryFileSystem trgMfs = new MemoryFileSystem();
-            CompilationResult result = compile(root, trgMfs, generatedFiles, generatedBeans, launchMode.getLaunchMode(),
+            CompilationResult result = compile(root, trgMfs, generatedFiles, launchMode.getLaunchMode(),
                     targetClassesPath);
-            register(trgMfs, generatedBeans, launchMode.getLaunchMode(), result, targetClassesPath);
+            register(trgMfs, generatedBeans, (className, data) -> {
+
+                IndexingUtil.indexClass(className, kogitoIndexer, combinedIndexBuildItem.getIndex(), kogitoIndex,
+                        Thread.currentThread().getContextClassLoader(),
+                        data);
+                return new GeneratedBeanBuildItem(className, data);
+            },
+                    launchMode.getLaunchMode(), result, targetClassesPath);
+
+            Index index = kogitoIndexer.complete();
+
+            generatePersistenceInfo(root, generatedBeans, CompositeIndex.create(combinedIndexBuildItem.getIndex(), index),
+                    launchMode, resource);
+
+            reflectiveClass.produce(
+                    new ReflectiveClassBuildItem(true, true, "org.kie.kogito.services.event.ProcessInstanceDataEvent"));
```

Comment:
Do you need method and field access?

Reply:
It is used by the jsonb Marshallers I I believe it is needed.